api/src/main/java/org/openmrs/module/fhirExtension/validators/DiagnosticReportRequestValidator.java

```diff
@@ -28,7 +26,10 @@ private void validateBasedOn(FhirDiagnosticReport fhirDiagnosticReport) {
         Patient patient = fhirDiagnosticReport.getSubject();
         Integer conceptId = fhirDiagnosticReport.getCode().getId();
         List<Order> allOrders = orderService.getAllOrdersByPatient(patient);
-        long matchingOrdersCount = allOrders.stream().filter(Order::isActive).filter((order) -> order.getConcept().getId().equals(conceptId)).count();
+        long matchingOrdersCount = allOrders.stream()
+                .filter(order -> !Order.FulfillerStatus.COMPLETED.equals(order.getFulfillerStatus()))
+                .filter(order -> !order.getVoided())
+                .filter((order) -> order.getConcept().getId().equals(conceptId)).count();
```

Comment:
So as per this implementation, it means that an order should always be present for a diagnostic report. 
As, If a report is uploaded without any order, we are verifying if the matching order is present or not.

Reply:
validation is done against the order